### ISSUE TITLE
Alpha Variants for Black and White in UnoCSS Radix Preset Render Incorrectly (Impacting Shadows and Backgrounds)

### ISSUE THREAD

**Issue by lafllamme** [endigma/unocss-preset-radix#25]
I’ve encountered an issue in the `unocss-preset-radix` package where the alpha variants for `blackA` and `whiteA` are not functioning as expected. This results in incorrect rendering, particularly noticeable with `box-shadow` and `background-color` utilities.

Example within Screenshot: Colors are not affected, only black and white.

![IMG 2024-12-24 02 29 32](https://github.com/user-attachments/assets/a2a9fa4c-3054-4f47-aca4-6c0e02f5e623)

#### Observations:
- The necessary CSS variables (e.g., `--un-preset-radix-black7A`, `--un-preset-radix-white8A`) are correctly defined in the `:root`.
- However, when applied via classes like `.bg-black-7A` or `.shadow-black-7A`, the output renders as fully opaque solid colors (`#000` and `#fff`), ignoring the alpha values. For example:

  ```css
  .bg-black,
  .bg-black-10A,
  .bg-black-11A,
  .bg-black-12A,
  .bg-black-1A,
  .bg-black-2A,
  .bg-black-3A,
  .bg-black-4A,
  .bg-black-5A,
  .bg-black-6A,
  .bg-black-7A,
  .bg-black-8A,
  .bg-black-9A {
    --un-bg-opacity: 1;
    background-color: rgb(0 0 0 / var(--un-bg-opacity)); /* Outputs #000 */
  }
  ```

- This issue affects `box-shadow` as well. Shadows like `shadow-black-7A` appear darker than expected due to the missing alpha transparency.

#### Example in Context:
I noticed this issue in my Vue component where I used `shadow-black-7A` for a subtle shadow effect. However, it rendered as a fully opaque shadow, making the design appear heavier than intended. Below is the relevant component snippet:

```vue
<!-- Popover.vue -->
<template>
  <PopoverRoot>
    <PopoverTrigger
      :class="
        useClsx(
          'shadow-[0_2px_10px] shadow-black-7A !bg-white',
          'hover:bg-mauve-3 focus:shadow-[0_0_0_2px]'
        )"
    >
      More info
    </PopoverTrigger>
  </PopoverRoot>
</template>
```

#### Expected Behavior:
Alpha levels for `blackA` and `whiteA` should apply the corresponding RGBA values from the root variables (e.g., `--un-preset-radix-black7A`). For instance:
- `.bg-black-7A` → `background-color: rgba(0, 0, 0, 0.5);`
- `.shadow-black-7A` → `box-shadow: 0px 2px 10px rgba(0, 0, 0, 0.5);`

#### Steps to Reproduce:
1. Use `unocss-preset-radix` in a project with UnoCSS.
2. Apply classes such as `.bg-black-7A` or `.shadow-black-7A`.
3. Inspect the generated CSS for these classes in devtools.

#### Feedback Requested:
1. Could this issue be due to how `unocss-preset-radix` maps `blackA` and `whiteA` in the preset? Other colors work as expected.
2. Are there better practices or optimizations I could use for my setup? For example, should I manually remap `blackA` and `whiteA` while waiting for a fix?

#### Additional Notes:
This issue appears to be isolated to the alpha scales of `black` and `white`. Other color scales like `blueA`, `grayA`, etc., work correctly. 

Would appreciate feedback and guidance on whether this can be addressed in the preset or if a temporary workaround is needed!

**Comment by endigma** [comment]
Thanks for the most thorough bug report I've ever seen, great work, honestly.

Ironically though I have a pretty good idea of what is causing this already, I'll fix it soon.

**Comment by lafllamme** [comment]
@endigma First of all, thank you for your time and hard work in providing this preset—it’s a perfect fit for the design system pattern I’m currently building. I also appreciate the clear bug report, as it helps me maintain transparency, especially since I’m not always 100% sure if an issue is related to a package itself or something else.

While preparing my patterns, I was going crazy trying to figure out why my shadows (and other elements) appeared so harsh... until I discovered the egg!

I took a look into the code but couldn’t pinpoint exactly where the issue lies. The variables seem to be calculated correctly, but it appears the mapping doesn’t actually use them. I also noticed that somewhere in the code, black and white are directly referenced as #000 and #FFF. Perhaps we might have overlooked proper mapping in this case?

Sincerely & happy xmas !

**Comment by endigma** [comment]
It's because `black` and `white` are just injected as supplementary unocss colors, which means opacity is written like `opacity-50 bg-black`. I'm looking into making the blackA and whiteA scales work as they do in radix while leaving the solids alone.

**Comment by endigma** [comment]
I've pushed [a fix](https://github.com/endigma/unocss-preset-radix/commit/02ec61c2b1f7964e969bfad09e418e12ab0ecd05), let me know if this fixes the issue for you.

**Comment by lafllamme** [comment]
Hi @endigma,

I reviewed the changes and they seem to fix the issue described. However, I’ve been unable to incorporate this specific commit into my project. I even tried forking the repo, building `./dist` manually, and adjusting `package.json`, but with no success.

Could you please share the exact steps I should follow to integrate this fix into my environment? Alternatively, if you’re able to release a new version that includes this commit, that would be greatly appreciated.

Thank you for all your hard work!

**Comment by endigma** [comment]
If that fixes it I'll make a release, for future reference though:

```json
"dependencies": {
  "package": "git://github.com/username/package.git#commit"
}
```

should work.

**Comment by endigma** [comment]
https://github.com/endigma/unocss-preset-radix/actions/runs/12503730134/job/34884499975

**Comment by endigma** [comment]
This is released in 2.5.6
